lib/flight_scheduler/job_registry.rb

```diff
@@ -130,6 +130,16 @@ def load
         Async.logger.warn("Invalid job loaded: #{job.errors.inspect}")
       end
     end
+    jobs.each do |job|
+      # NOTE: There is an edge case where the "insufficient script" would have otherwise
+      # ran before the reboot, but not after. This can occur if the default reason
+      # in the scheduler is changed to anything other than 'Resources'
+      #
+      # This does not happen in the FiFoScheduler and BackfillingScheduler, so can
+      # be ignored for the time being.
+      # Consider revisiting when this no longer holds
+      job.partition.script_runner.insufficient if job.reason_pending == 'Resources'
+    end
```

Comment:
I don't understand why we need this.  What would be wrong with simply waiting until we run the next allocation loop?

Reply:
Because an `insufficient` request could be in the denouncing loop whilst the service is being reloaded, which could cause it to be missed. The allocation loop does not run at regular intervals, so we can not assume it will pick up the missed message.

Reply:
So that's the problem that we're working around here.  We need the allocation loop to run at regular intervals.  We'll get to that in another PR.